src/game/stats.c

```diff
@@ -419,6 +421,10 @@ void Stats_ShowTotal(const char *filename)
         total_max_pickup_count += stats->max_pickup_count;
     }
 
+    // Check for bonus level unlock.
+    g_GameInfo.bonus_level_unlock =
+        total_secret_count >= total_max_secret_count;
```

Comment:
Let's try to avoid mutating global state in functions like `Show`, `Get` etc.
To be able to reuse `total_Secret_count`, consider making a function called `Stats_Compute` that fills a struct with game statistics, and calling it twice, once in `Stats_Show`, and once to update the `g_GameInfo`. The computational cost is low, and I believe the maintenance becomes easier.

Reply:
Good ideas. Lmk what you think of what I pushed.